src/Stripe.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -81,7 +85,7 @@ private static void TryAddType(IServiceCollection collection, Type type)
             {
                 if (parameter.ParameterType == typeof(IStripeClient))
                 {
-                    collection.TryAdd(ServiceDescriptor.Transient(type, type));
+                    collection.TryAdd(ServiceDescriptor.Singleton(type, type));
```

Comment:
The HttpClient returned by IHttpClientFactory can be hold onto forever. Internally it keeps the connection up-to-date.